server/plugins/api/Twitter.ts

```diff
@@ -1,22 +1,71 @@
 import { APIReturn } from "./APIReturn";
 import { GenericNotifier } from "./GenericNotifier";
 
+import * as TwitterAPI from "twit";
+
 export default class Twitter implements GenericNotifier<{}> {
 
 	public async setup(): Promise<void> {
-		return;
+		const consumerKey = process.env.TWITTER_CONSUMER_KEY;
+		const consumerSecret = process.env.TWITTER_CONSUMER_SECRET;
+		const accessToken = process.env.TWITTER_HACKGT_ACCESS_TOKEN;
+		const accessTokenSecret = process.env.TWITTER_HACKGT_ACCESS_TOKEN_SECRET;
+		let envFlag: boolean = false;
+		if (!consumerKey) {
+			console.log("Missing Twitter consumer_key env var");
+			envFlag = true;
+		}
+		if (!consumerSecret) {
+			console.log("Missing Twitter consumer_secret env var");
+			envFlag = true;
+		}
+		if (!accessToken) {
+			console.log("Missing Twitter access_token env var");
+			envFlag = true;
+		}
+		if (!accessTokenSecret) {
+			console.log("Missing Twitter access_token_secret env var");
+			envFlag = true;
+		}
+
+		if ( envFlag ) {
+			throw new Error("Twitter env vars missing, aborting...");
+		}
+
+		return; // No setup currently
 	}
 
 	public async check(config: any): Promise<{}> {
-		return {};
+		return {}; // No config currently
 	}
 
 	public async sendMessage(message: string, config: {}): Promise<[APIReturn]> {
-		return [{
-			error: false,
-			key: "Twitter",
-			message: "Twitter success"
-		}];
+		const client = new TwitterAPI({
+			consumer_key: (process.env.TWITTER_CONSUMER_KEY as string),
+			consumer_secret: (process.env.TWITTER_CONSUMER_SECRET as string),
+			access_token: (process.env.TWITTER_HACKGT_ACCESS_TOKEN as string),
+			access_token_secret: (process.env.TWITTER_HACKGT_ACCESS_TOKEN_SECRET as string)
+		});
+		const params = { status: message };
+		const res = await new Promise((resolve, reject) => {
```

Comment:
Can you fix the asset by using `res: APIReturn`?

Reply:
Nah, throws the error that I can't cast `{}` to `APIReturn`.

Reply:
It works if you first create a var `const promise: Promise<APIReturn> = new Promise(...` then await on it.